File: core/src/main/java/com/eloja/core/entity/Usuario.java
package com.eloja.core.entity;


import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String senha;

    @OneToOne(cascade = CascadeType.PERSIST)
    private Fisica fisica;

    @OneToOne(cascade = CascadeType.PERSIST)
    private Juridica juridica;


}

File: frontend-react/src/models/Usuario.ts
import Fisica from './Fisica';
import Juridica from './Juridica';

class Usuario {
  id?: number;
  email?: string;
  senha?: string;
  fisica?: Fisica;
  juridica?: Juridica;

  constructor(email?: string, senha?: string, fisica?: Fisica, juridica?: Juridica, id?: number) {
    this.id = id;
    this.email = email;
    this.senha = senha;
    this.fisica = fisica;
    this.juridica = juridica;
  }
}

export default Usuario;
